engine/Object.js

```diff
@@ -106,22 +106,22 @@ export class Object{
 	CompactPathString(otherPath){
 		var globalPathStr = null;
 		var relativePathStr = null;
-		
+
 		if (otherPath.isRelative) {
 			relativePathStr = otherPath.componentsString;
 			globalPathStr = this.path.PathByAppendingPath(otherPath).componentsString;
-		} 
+		}
 		else {
 			var relativePath = this.ConvertPathToRelative(otherPath);
 			relativePathStr = relativePath.componentsString;
 			globalPathStr = otherPath.componentsString;
 		}
 
-		if (relativePathStr.Length < globalPathStr.Length) 
+		if (relativePathStr.length < globalPathStr.length)
```

Comment:
In hindsight, `path.length` should probably be `path.Length`. Let's keep it like that and fix it later.